cmd/apprepository-controller/main.go

```diff
@@ -87,18 +159,38 @@ func main() {
 	}
 }
 
-func init() {
-	flag.StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
-	flag.StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
-	flag.StringVar(&repoSyncImage, "repo-sync-image", "quay.io/helmpack/chart-repo:latest", "container repo/image to use in CronJobs")
-	flag.StringVar(&repoSyncCommand, "repo-sync-cmd", "/chart-repo", "command used to sync/delete repos for repo-sync-image")
-	flag.StringVar(&namespace, "namespace", "kubeapps", "Namespace to discover AppRepository resources")
-	flag.BoolVar(&reposPerNamespace, "repos-per-namespace", true, "Defaults to watch for repos in all namespaces. Switch to false to watch only the configured namespace.")
-	flag.StringVar(&dbURL, "database-url", "localhost", "Database URL")
-	flag.StringVar(&dbUser, "database-user", "root", "Database user")
-	flag.StringVar(&dbName, "database-name", "charts", "Database name")
-	flag.StringVar(&dbSecretName, "database-secret-name", "kubeapps-db", "Kubernetes secret name for database credentials")
-	flag.StringVar(&dbSecretKey, "database-secret-key", "postgresql-root-password", "Kubernetes secret key used for database credentials")
-	flag.StringVar(&userAgentComment, "user-agent-comment", "", "UserAgent comment used during outbound requests")
-	flag.StringVar(&crontab, "crontab", "*/10 * * * *", "CronTab to specify schedule")
+// getImagePullSecretsRefs gets the []LocalObjectReference of Secrets from the
+// comma separated list passed in the repoSyncImagePullSecrets arg
+func getImagePullSecretsRefs(imagePullSecretsRefsParam arrayFlags) []corev1.LocalObjectReference {
+	var imagePullSecretsRefs []corev1.LocalObjectReference
+
+	// if no repoSyncImagePullSecrets arg passed, return nil, as usual
+	if len(imagePullSecretsRefsParam) == 0 {
+		return nil
+	}
+
+	// getting and appending a []LocalObjectReference for each ImagePullSecret passed
+	for _, imagePullSecretName := range imagePullSecretsRefsParam {
+		imagePullSecretsRefs = append(imagePullSecretsRefs, []corev1.LocalObjectReference{
+			{Name: imagePullSecretName},
+		}...)
+	}
+	return imagePullSecretsRefs
+}
+
+// just for being compliant with the interface
+func (i *arrayFlags) String() string {
+	return fmt.Sprint(*i)
+}
+
+// if the flag arg it's a comma-separated list, split it by "," and append
+// otherwise append it directly (dups allowed)
+func (i *arrayFlags) Set(value string) error {
+	if strings.Contains(value, ",") {
+		*i = append(*i, strings.Split(strings.TrimSpace(value), ",")...)
+
+	} else {
+		*i = append(*i, strings.TrimSpace(value))
+	}
+	return nil
```

Comment:
So this is a bit of custom code we'll need to test and maintain to support both multiple args as well as comma-separated args. When you find yourself doing this, it's often a good idea to step back and ask whether others have already solved this. In this case, we've used spf13/pflag as a drop-in replacement for go's standard flag lib. It actually provides this functionality out of the box. See https://pkg.go.dev/github.com/spf13/pflag#StringSlice .

Reply:
yep, since you are changing this much how the flags are set, we can use a more complete library like the one Michael suggests.